### ISSUE TITLE
Concern about the meaning of sdf 

### ISSUE THREAD

**Issue by hirotong** [dazinovic/neural-rgbd-surface-reconstruction#12]
Hi, @dazinovic 
Thx for sharing your great work!
I'm confused about the meaning of **sdf** used in your code. I hope you can help me figure it out.
https://github.com/dazinovic/neural-rgbd-surface-reconstruction/blob/01782741271b5ace1083bfc4aadd787c97b31836/losses.py#L31-L38
As in this function, it seems that **sdf** is a ratio between the actual distance and the truncation value.
While, when it turns to this render function, 
https://github.com/dazinovic/neural-rgbd-surface-reconstruction/blob/01782741271b5ace1083bfc4aadd787c97b31836/optimize.py#L172-L183
it seems that **sdf** represents a distance in **normalized space**. 
So, if we use the second meaning which is **sdf is a distance in normalized space**, should the sdf loss function change to this?
```python
def get_sdf_loss(z_vals, target_d, predicted_sdf, truncation, loss_type):

    front_mask, sdf_mask, fs_weight, sdf_weight = get_masks(z_vals, target_d, truncation)

    fs_loss = compute_loss(predicted_sdf * front_mask, truncation * tf.ones_like(predicted_sdf) * front_mask, loss_type) * fs_weight
    sdf_loss = compute_loss((z_vals + predicted_sdf) * sdf_mask, target_d * sdf_mask, loss_type) * sdf_weight

    return fs_loss, sdf_loss
```

**Comment by dazinovic** [comment]
The predicted SDF is always normalized between -1 (behind the surface) and +1 (in front of the surface). In the loss, it needs to be multiplied by truncation since all the other variables are in metric space (possibly scaled by sc_factor from the config file).

Let's say the depth map pixel has a value of 0.8, the sample in z_vals has a value of 0.775 (difference of 2.5 cm). Then the predicted SDF needs to be 0.025 / truncation (0.5 with a truncation of 5 cm). So to turn this back into metric space and apply the loss, you need to multiple the predicted SDF by the truncation size.

**Comment by hirotong** [comment]
Hi, Dazinovic,
Thanks for your reply. That makes sense.
But I still have a question about the **weights** in function **sdf2weights**.
Why do you use the ratio sdf/truncation as input? Intuitively, they're not in the same space.

**Comment by chensjtu** [comment]
> Hi, Dazinovic,
Thanks for your reply. That makes sense.
But I still have a question about the weights in function sdf2weights.
Why do you use the ratio sdf/truncation as input? Intuitively, they're not in the same space.

The author only uses truncation without scfactor in sdf2weight, which is confusing.

**Comment by dazinovic** [comment]
> Yes. The 'truncation' in sdf2weights function is more like a scale factor. As said in the paper, it controls how fast the weight falls to zero. It's not actually the truncation distance but happens to get a good result with the same value as "truncation distance". Those are only my suspects. Hopefully, the author will give a better comment about it.

You are right about that. We used a fixed scaling of 20.0 inside sdf2weight in our experiments (decided to use that after trying out a few numbers), but noticed last minute before submitting the paper that it can also be written more elegantly by using the truncation distance.

**Comment by HtForBetterLife** [comment]
hi, @hirotong 
Do you know the meaning of 'mask' in sdf2weights() ?

**Comment by dazinovic** [comment]
> 

A ray might pass through multiple surfaces. That's why the samples past the first truncation region are masked out.

**Comment by HtForBetterLife** [comment]
> 

I got it! Thanks for your reply.
There is another question:
Why do you use “signs = sdf[:, 1:] * sdf[:, :-1]” to generate weights instead of sdf directly?
